common/v2/services/Store/BalanceService.tsx

```diff
@@ -81,13 +92,15 @@ const getTokenBalances = (
   provider: ProviderHandler,
   address: TAddress,
   tokens: StoreAsset[]
-): BalanceMap => {
-  return tokens.reduce(async (balances, token) => {
-    return {
-      ...balances,
-      [token.contractAddress as TAddress]: await provider.getRawTokenBalance(address, token)
-    };
-  }, {});
+): Promise<BalanceMap> => {
+  return tokens
+    .reduce<Promise<EthScanBalanceMap>>(async (balances, token) => {
+      return Promise.resolve({
```

Comment:
I don't think you need the `Promise.resolve` here (since the function is already a `Promise`. Other than that it looks good to me.